File: mobile/apps/photos/plugins/ente_photos_platform/ios/Flutter/PhotosPlatformPlugin.swift
@preconcurrency import Flutter
import Foundation

@MainActor
public final class PhotosPlatformPlugin: NSObject, @preconcurrency FlutterPlugin {
    private let deviceHealthAdapter: DeviceHealthChannelAdapter
    private let deviceTrashAdapter: DeviceTrashChannelAdapter

    private init(registrar: FlutterPluginRegistrar) {
        deviceHealthAdapter = DeviceHealthChannelAdapter(registrar: registrar)
        deviceTrashAdapter = DeviceTrashChannelAdapter(registrar: registrar)
        super.init()
    }

    public static func register(with registrar: FlutterPluginRegistrar) {
        registrar.publish(PhotosPlatformPlugin(registrar: registrar))
    }

    public func detachFromEngine(for registrar: FlutterPluginRegistrar) {
        deviceTrashAdapter.detach()
        deviceHealthAdapter.detach()
    }
}

File: mobile/apps/photos/plugins/ente_photos_platform/android/src/main/kotlin/io/ente/photos/platform/PhotosPlatformPlugin.kt
package io.ente.photos.platform

import io.ente.photos.platform.flutter.DeviceHealthChannelAdapter
import io.ente.photos.platform.flutter.DeviceTrashChannelAdapter
import io.flutter.embedding.engine.plugins.FlutterPlugin

class PhotosPlatformPlugin : FlutterPlugin {
    private val deviceHealthAdapter = DeviceHealthChannelAdapter()
    private val deviceTrashAdapter = DeviceTrashChannelAdapter()

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        deviceHealthAdapter.attach(binding)
        deviceTrashAdapter.attach(binding)
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        deviceTrashAdapter.detach()
        deviceHealthAdapter.detach()
    }
}
